src/client/address_book/impl.cc

```diff
@@ -528,6 +528,68 @@ void AddressBook::Stop() {
   m_Subscribers.clear();
 }
 
+BookEntry::BookEntry(
+    const std::string& host,
+    const kovri::core::IdentHash& address) try : m_Host(host),
+                                                 m_Address(address)
+  {
+    if (m_Host.empty())
+      throw std::invalid_argument("AddressBook: empty entry hostname");
+  }
+catch (...)
+  {
+    kovri::core::Exception ex(__func__);
+    ex.Dispatch();
+    throw;
+  }
+
+BookEntry::BookEntry(
+    const std::string& host,
+    const std::string& address) try : m_Host(host)
+  {
+    if (m_Host.empty())
+      throw std::invalid_argument("AddressBook: empty entry hostname");
```

Comment:
Would you NACK adding a similar check for an empty address?
For example:

```
if (address.empty())
  throw std::invalid_argument("AddressBook: empty entry address");
```

IIUC now, this check is more efficient than the call to `ident.FromBase64`. Please correct me if I'm wrong.

Reply:
That depends on the case. I wouldn't concern myself with that right now for the remainder of this PR.

Reply:
Alright, will leave it out for now.